libs/datamodel/connectors/datamodel-connector/src/walker_ext_traits.rs

```diff
@@ -28,13 +26,64 @@ impl<'ast> IndexWalkerExt<'ast> for IndexWalker<'ast, '_> {
 }
 
-impl<'db> DefaultValueExt<'db> for DefaultValueWalker<'_, 'db> {
-    fn constraint_name(self, connector: &dyn Connector) -> Cow<'db, str> {
+impl<'ast> DefaultValueExt<'ast> for DefaultValueWalker<'ast, '_> {
+    fn dml_default_kind(self) -> dml::default_value::DefaultKind {
+        use dml::{
+            default_value::{DefaultKind, ValueGenerator},
+            prisma_value::PrismaValue,
+        };
+
+        // This has all been validated in parser-database, so unwrapping is always safe.
+        match self.value() {
+            ast::Expression::Function(funcname, args, _) if funcname == "dbgenerated" => {
+                DefaultKind::Expression(ValueGenerator::new_dbgenerated(
+                    args.get(0)
+                        .and_then(|arg| arg.as_string_value())
+                        .map(|(val, _)| val.to_owned())
+                        .unwrap_or_else(String::new),
+                ))
+            }
+            ast::Expression::Function(funcname, _args, _) if funcname == "autoincrement" => {
+                DefaultKind::Expression(ValueGenerator::new_autoincrement())
+            }
+            ast::Expression::Function(funcname, _args, _) if funcname == "uuid" => {
+                DefaultKind::Expression(ValueGenerator::new_uuid())
+            }
+            ast::Expression::Function(funcname, _args, _) if funcname == "cuid" => {
+                DefaultKind::Expression(ValueGenerator::new_cuid())
+            }
+            ast::Expression::Function(funcname, _args, _) if funcname == "now" => {
+                DefaultKind::Expression(ValueGenerator::new_now())
+            }
+            ast::Expression::NumericValue(num, _) => match self.field().scalar_type() {
+                Some(ScalarType::Int) => DefaultKind::Single(PrismaValue::Int(num.parse().unwrap())),
+                Some(ScalarType::BigInt) => DefaultKind::Single(PrismaValue::BigInt(num.parse().unwrap())),
+                Some(ScalarType::Float) => DefaultKind::Single(PrismaValue::Float(num.parse().unwrap())),
+                Some(ScalarType::Decimal) => DefaultKind::Single(PrismaValue::Float(num.parse().unwrap())),
+                other => unreachable!("{:?}", other),
+            },
+            ast::Expression::BooleanValue(b, _) => DefaultKind::Single(PrismaValue::Boolean(b.parse().unwrap())),
+            ast::Expression::ConstantValue(v, _) => DefaultKind::Single(PrismaValue::Enum(v.to_owned())),
+            ast::Expression::StringValue(v, _) => match self.field().scalar_type() {
+                Some(ScalarType::DateTime) => DefaultKind::Single(PrismaValue::DateTime(v.parse().unwrap())),
+                Some(ScalarType::String) => DefaultKind::Single(PrismaValue::String(v.parse().unwrap())),
+                Some(ScalarType::Json) => DefaultKind::Single(PrismaValue::Json(v.parse().unwrap())),
+                Some(ScalarType::Decimal) => DefaultKind::Single(PrismaValue::Float(v.parse().unwrap())),
```

Comment:
Do we accept decimals as numerics and strings?

Reply:
So I think I just ported the existing logic, I was also surprised. I'm afraid of breakage if we get more strict, but I agree it probably should be only numeric literals.